src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -38,107 +39,66 @@ internal class ProjectBuildManager
             // don't actually run the compiler
             { PropertyNames.SkipCompilerExecution, bool.TrueString },
 
-            { PropertyNames.ContinueOnError, PropertyValues.ErrorAndContinue }
+            { PropertyNames.ContinueOnError, PropertyValues.ErrorAndContinue },
+
+            // this ensures that the parent project's configuration and platform will be used for
+            // referenced projects. So, setting Configuration=Release will also cause any project
+            // references to also be built with Configuration=Release. This is necessary for getting
+            // the correct output path from project references.
```

Comment:
```suggestion
            // a more-likely-to-be-correct output path from project references.
```

? It's correct for the started-with-a-single-project case, and correct in the solution case if there are no solution-configuration shenanigans, but possibly wrong if there are.

Reply:
That comment tweak expresses the right amount of paranoia. We aren't doing solution-configuration-shenanigans today, but might in the future.

Reply:
> the right amount of paranoia

#lifegoals